apps/mobile/src/components/TradeChecklistFlow/components/CalculateAmountFlow/atoms.ts

```diff
@@ -230,6 +242,30 @@ export const calculateBtcValueOnFiatAmountChangeActionAtom = atom(
   }
 )
 
+export const refreshCurrentBtcPriceActionAtom = atom(null, (get, set) => {
+  const currency =
+    get(selectedCurrencyCodeAtom) ??
+    get(tradeOrOriginOfferCurrencyAtom) ??
+    'USD'
+  const btcInputValue = get(btcInputValueAtom)
+
+  return pipe(
+    set(refreshBtcPriceActionAtom, currency),
+    T.map(() => {
+      set(
+        tradeBtcPriceAtom,
+        (prev) => get(btcPriceForOfferWithStateAtom)?.btcPrice ?? prev
+      )
+      // we need to recalculate amount of fiat based on new btc price
+      set(calculateFiatValueOnBtcAmountChangeActionAtom, {
```

Comment:
To me this seems like an antipattern. 

Atoms are reactive and they should react on other atoms' changes.

Calling set atom with values that were retrieved from other atoms is not the right approach (for example `btcAmount` parameter). 

The proper way how to do this would be to move this logic into atom's getter and just set atom value here.

I understand that it would be pretty significant logic change to fix this here, but it adds unnecessary complexity that will cause small changes being harder and harder to implement. 

Please add todo and we'll fix this next time when (if) we revisit this feature and have more time...

Reply:
Yeah you are right, I was also thinking the same when writing this code, but did not want to change this calculate function to be able to deliver the feature, but I totally agree with you

Reply:
Let's add the todo